ocaml/typing/solver.ml

```diff
@@ -206,7 +218,8 @@ module Solver_mono (C : Lattices_mono) = struct
     | Amodemeet (a, vs) ->
       Amodemeet (C.apply dst morph a, List.map (apply_morphvar dst morph) vs)
 
-  let update_lower (type a) ?log (obj : a C.obj) v a =
+  (** Arguments not checked; must maintain INVARIANT *)
```

Comment:
Not exactly sure what this comment means. Who must maintain the invariant? The caller of this function? or the function itself?

Ditto for other similar comments.